pagopa/io-app
# chore: [IA-424] Upgrade @pagopa/io-pagopa-commons

## Short description
While working on the payment receipts and the _paid_ status, I've found that we were lagging behind with the library in the title and decided to try the major leap: it works 🥳 .
It resulted in minor updates (many files but basically imports only) and some dead code removed (see`ts/utils/amounts.ts`)

To let my fellows do it more confidently the next time, I've added a full-fledged e2e test covering the happy path of a payment.

## List of changes proposed in this pull request
- `@pagopa/io-pagopa-commons:2.0.2`
- `mockttp:2.4.0` (used in tests only)
- add [new e2e test](https://github.com/pagopa/io-app/compare/IA-424-upgrade-pagopa-commons?expand=1#diff-ce0c408cb6b2f7d5355ce19f5d3851f4f4a5a015f03cb3b7aff86438b5dc6160R121) for successful payment 

## How to test
To see the e2e test running on your machine, please use [this version](https://github.com/pagopa/io-dev-api-server/pull/90) of the dev-server and follow the instructions in `./TESTING_E2E.md`.
